sycl/doc/extensions/experimental/sycl_ext_oneapi_graph.asciidoc

```diff
@@ -1167,6 +1192,74 @@ Exceptions:
   `property::graph::updatable` was not set when the executable graph was created.
 * Throws with error code `invalid` if any node in `nodes` is not part of the
   graph.
+
+|
+[source, c++]
+----
+void
+update(const command_graph<graph_state::modifiable>& graph);
+----
+
+|Updates all of the nodes in the executable graph with parameters from a
+topologically identical modifiable graph. A topologically identical graph is one
+with the same structure of nodes and edges. Additionally, all nodes and edges
+must be added in the same order, and all Graph-related API calls and recorded
+queue submissions must occur in the same order.
```

Comment:
I agree it is a little underspecified. The motivations for this are:

- Some backends (aka CUDA) have similar requirements for their capture APIs. We don't currently use the capture API for CUDA in the backend, but it is conceivable that we may want to in the future.
- It is difficult to define exactly what topologically identical means, consider this example:

The implementation cannot reliably in all cases distinguish between:

```
NodeA        NodeC
  |             |
NodeB        NodeD
```

from:
```
NodeC        NodeA
  |             |
NodeD        NodeB
```
If enough is the same between nodes (argument number, sizes etc.). This is also potentially exacerbated if we start allowing kernel functions to be swapped, since that removes another thing we can compare. So it seems if we cannot be 100% certain of being able to report errors that it is safest to mandate that the APIs are called in the same order (and that is likely the motivation CUDA has for it as well).

This may be too far reaching, in that you are right probably dynamic_parameters don't matter for this, but likely any calls which modify the graph topology would still need this requirement so that could be specified in more detail.

Reply:
The two graphs you show above in your example look "topologically identical" to me.  I guess you are saying that it's hard for the implementation to determine this because the nodes are in a different order?  Thus, the implementation would need to sort them somehow in order to determine which nodes correspond?  I can see how that might be difficult, or computationally expensive.

My top-level comment is that we need some more precise rule about what the requirement is.  For example, maybe the spec needs to list exactly which APIs must be called in the same order.  Related, is my other comment about clarifying exactly what we mean by "same command".